src/view/components/EditLoanEntry/index.tsx

```diff
@@ -0,0 +1,191 @@
+import * as React from 'react';
+import { Content, List, View, ListItem, Text, Grid, Row, Col, H1, Button, Input, Item } from 'native-base';
+
+import { Farmer } from '../../../store/modules/farmer/types';
+import { LoanEntry } from '../../../store/modules/loan/types';
+import { Route } from '../../navigation/routes';
+
+import { MapStateToProps, MapDispatchToProps, connect } from 'react-redux';
+import navActions from '../../../store/modules/nav/actions';
+import { InjectedFabProps } from '../../hoc/PageComposer/FabPage/index';
+import Composer from '../../hoc/PageComposer/index';
+import { State, ThunkUpdateRow, StoreRow } from '../../../store/types';
+
+import loanThunks from '../../../store/modules/loan/thunks';
+
+import Styles from './style';
+import { getActiveFarmer } from '../../../store/modules/farmer/selectors';
+import { getActiveLoanEntry } from '../../../store/modules/loan/selectors';
+import * as moment from 'moment';
+
+
+interface OwnPropsType {
+}
+
+/** EditLoanEntry DispatchPropsType */
+interface DispatchPropsType {
+  navigate(route: Route): void;
+  goBack(): void;
+  updateLoanEntry(newEntry: ThunkUpdateRow<LoanEntry>): void;
+}
+
+/** EditLoanEntry StorePropsType */
+interface StorePropsType {
+  farmer: Farmer;
+  loanEntry: StoreRow<LoanEntry>;
+}
+
+type NestedPropsType = StorePropsType & DispatchPropsType & OwnPropsType;
+
+/** EditLoanEntry PropsType */
+type PropsType = InjectedFabProps & NestedPropsType;
+
+/** EditLoanEntry OwnStateType */
+interface OwnStateType {
+  amountOfLoan: number;
+}
+
+/**
+ * Button color
+ */
+type ButtonColor = 'PRIMARY' | 'INFO';
+
+/**
+ * Page for EditLoanEntry
+ */
+class EditLoanEntry extends React.Component<PropsType, OwnStateType> {
+
+  constructor(props: PropsType) {
+    super(props);
+    this.state = {
+      amountOfLoan: this.props.loanEntry.amount,
+    };
+  }
+
+  /** Create page buttons */
+  private renderCancelButton = () => this.renderButton('Cancel', 'INFO', this.onCancelPress);
+  private renderSaveButton = () => this.renderButton('Save', 'PRIMARY', this.onSavePress);
+
+  /** Handle pressing cancel button */
+  private onCancelPress = () => this.props.goBack();
+
+  /** Handle pressing save button */
+  private onSavePress = () => {
+    let newEntry: ThunkUpdateRow<LoanEntry> = {
+      uuid: this.props.loanEntry.uuid,
+      toPersonUuid: this.props.loanEntry.toPersonUuid,
+      fromPersonUuid: this.props.loanEntry.fromPersonUuid,
+      amount: this.state.amountOfLoan,
+      currency: 'UGX',
+    };
+    this.props.updateLoanEntry(newEntry);
+    this.props.navigate(Route.FARMER);
+  }
+
+  /**
+   * Handle entry changes, update local state
+   */
+  private onAmountChange = (newAmount: string) => this.setState(state => ({ amountOfLoan: parseFloat(newAmount) }));
```

Comment:
We need to validate input here like we did in `AddMilkEntry`.

Reply:
Input validation added